app/assets/javascripts/pretty-text/engines/discourse-markdown/anchor.js

```diff
@@ -37,15 +39,7 @@ export function setup(helper) {
           .replace(/^-+/, "")
           .replace(/-+$/, "");
 
-        if (slug.length === 0) {
-          slug = state.tokens[idx + 1].content
```

Comment:
Did we mean to remove the special characters code? The description doesn't mention this.

Reply:
Yes, that code used to generate really bad names, which seemed completely random. Unfortunately, a better solution cannot be implemented because we do some sanitization of special characters on the server side.

Reply:
Sounds good.